src/main/java/Deadline.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * This is the Deadline class. It contains TaskList objects that are assigned deadline.
+ * @author Hua Lun
+ */
+import java.util.Date;
+public class Deadline extends TaskList {
```

Comment:
A deadline isn't a list, tasklist is meant to store the tasks, deadline should inherit from something like Task.